src/sqlfluff/utils/reflow/sequence.py

```diff
@@ -74,6 +79,58 @@ def get_fixes(self) -> List[LintFix]:
         """Get the current fix buffer."""
         return self.embodied_fixes
 
+    def get_results(self) -> List[LintResult]:
+        """Generate a list of LintResult from the current fix buffer.
+
+        For some fixes, we do a bit of a shuffle with the anchor so
+        that we reference the most sensible position in the source file
+        to make sense to the user. For example if we've due to create
+        something *after* a segment, then we'll hunt forward to the
+        segment *after that*, so that the position that appears in the
+        CLI is the position between them.
+
+        NOTE: This generates one result per fix. Depending on your use
+        case, this may not always be appropriate. In those cases, call
+        `ReflowSequence.get_fixes()` and construct :obj:`LintResult`
+        objects directly.
+        """
+        results = []
+        segments = None
+        for fix in self.get_fixes():
+            # Anchoring on the segment AFTER makes is a lot more
+            # understandable to the user, that's why we do a little
+            # shuffle here.
+            if fix.edit_type == "create_after" or (
+                fix.edit_type == "replace"
+                and "".join(
+                    seg.raw for seg in cast(List[BaseSegment], fix.edit)
+                ).startswith(fix.anchor.raw)
+            ):
+                if not segments:
+                    segments = list(
+                        chain.from_iterable(elem.segments for elem in self.elements)
+                    )
+
+                target = fix.anchor.segments[-1] if fix.anchor.segments else fix.anchor
+                try:
+                    idx = segments.index(target)
+                except ValueError:
+                    # the anchor isn't there any more, it's a good anchor.
```

Comment:
This comment confuses me. If it's not there, how is it good? 🤔

Reply:
I've updated the comment to:
```python
                    # If the anchor for the fix wasn't found, then that
                    # means that it has been replaced or removed itself.

                    # The most likely scenario for this is where the anchor
                    # for the fix is also part of the problem - therefore
                    # using it as the position for the linting result is ok.
                    # We'll set the idx to None to signal this so that in
                    # the next clause we just set the anchor of the linting
                    # result to the anchor of the fix.
```